From ae6763b34e41b89301230f5dd7796c9fc88933ba Mon Sep 17 00:00:00 2001
From: Vincent Abriou <vincent.abriou@st.com>
Date: Wed, 12 Apr 2017 15:27:28 +0200
Subject: [PATCH 2/3] mali: replace <asm/uaccess.h> by <linux/uaccess.h>

This fix the build issue due to the fact that copy_to_user and
copy_from_user has been replaces by raw_copy_to_user and
raw_copy_from_user.

Build issue introduced by:
commit 4de5b63e7 "arm: switch to RAW_COPY_USER"

Change-Id: Ied0e2f77e3aebe71daa545062315aa02e4be58f7
Signed-off-by: Vincent Abriou <vincent.abriou@st.com>
Reviewed-on: https://gerrit.st.com/68527
Reviewed-by: CITOOLS <smet-aci-reviews@lists.codex.cro.st.com>
Reviewed-by: Christophe PRIOUZEAU <christophe.priouzeau@st.com>
---
 src/devicedrv/mali/linux/mali_kernel_sysfs.c    | 2 +-
 src/devicedrv/mali/linux/mali_memory_dma_buf.c  | 2 +-
 src/devicedrv/mali/linux/mali_osk_mali.c        | 2 +-
 src/devicedrv/mali/linux/mali_osk_misc.c        | 2 +-
 src/devicedrv/mali/linux/mali_osk_specific.h    | 2 +-
 src/devicedrv/mali/linux/mali_ukk_core.c        | 2 +-
 src/devicedrv/mali/linux/mali_ukk_gp.c          | 2 +-
 src/devicedrv/mali/linux/mali_ukk_mem.c         | 2 +-
 src/devicedrv/mali/linux/mali_ukk_pp.c          | 2 +-
 src/devicedrv/mali/linux/mali_ukk_profiling.c   | 2 +-
 src/devicedrv/mali/linux/mali_ukk_soft_job.c    | 2 +-
 src/devicedrv/mali/linux/mali_ukk_timeline.c    | 2 +-
 src/devicedrv/mali/linux/mali_ukk_vsync.c       | 2 +-
 src/devicedrv/ump/linux/ump_kernel_linux.c      | 2 +-
 src/devicedrv/ump/linux/ump_osk_low_level_mem.c | 2 +-
 src/devicedrv/ump/linux/ump_ukk_ref_wrappers.c  | 2 +-
 src/devicedrv/ump/linux/ump_ukk_wrappers.c      | 2 +-
 src/devicedrv/umplock/umplock_driver.c          | 2 +-
 18 files changed, 18 insertions(+), 18 deletions(-)

diff --git a/src/devicedrv/mali/linux/mali_kernel_sysfs.c b/src/devicedrv/mali/linux/mali_kernel_sysfs.c
index 3b50af6..e6c92ac 100755
--- a/src/devicedrv/mali/linux/mali_kernel_sysfs.c
+++ b/src/devicedrv/mali/linux/mali_kernel_sysfs.c
@@ -26,7 +26,7 @@
 
 #include <linux/seq_file.h>
 #include <linux/debugfs.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <linux/module.h>
 #include <linux/mali/mali_utgard.h>
 #include "mali_kernel_sysfs.h"
diff --git a/src/devicedrv/mali/linux/mali_memory_dma_buf.c b/src/devicedrv/mali/linux/mali_memory_dma_buf.c
index f0d1e07..4a38c53 100755
--- a/src/devicedrv/mali/linux/mali_memory_dma_buf.c
+++ b/src/devicedrv/mali/linux/mali_memory_dma_buf.c
@@ -9,7 +9,7 @@
  */
 
 #include <linux/fs.h>      /* file system operations */
-#include <asm/uaccess.h>        /* user space access */
+#include <linux/uaccess.h>        /* user space access */
 #include <linux/dma-buf.h>
 #include <linux/scatterlist.h>
 #include <linux/rbtree.h>
diff --git a/src/devicedrv/mali/linux/mali_osk_mali.c b/src/devicedrv/mali/linux/mali_osk_mali.c
index 23b2c17..3835222 100755
--- a/src/devicedrv/mali/linux/mali_osk_mali.c
+++ b/src/devicedrv/mali/linux/mali_osk_mali.c
@@ -13,7 +13,7 @@
  * Implementation of the OS abstraction layer which is specific for the Mali kernel device driver
  */
 #include <linux/kernel.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <linux/platform_device.h>
 #include <linux/mali/mali_utgard.h>
 #include <linux/of.h>
diff --git a/src/devicedrv/mali/linux/mali_osk_misc.c b/src/devicedrv/mali/linux/mali_osk_misc.c
index 7dda283..be75eb9 100755
--- a/src/devicedrv/mali/linux/mali_osk_misc.c
+++ b/src/devicedrv/mali/linux/mali_osk_misc.c
@@ -13,7 +13,7 @@
  * Implementation of the OS abstraction layer for the kernel device driver
  */
 #include <linux/kernel.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <asm/cacheflush.h>
 #include <linux/sched.h>
 #include <linux/seq_file.h>
diff --git a/src/devicedrv/mali/linux/mali_osk_specific.h b/src/devicedrv/mali/linux/mali_osk_specific.h
index adcca29..199363e 100755
--- a/src/devicedrv/mali/linux/mali_osk_specific.h
+++ b/src/devicedrv/mali/linux/mali_osk_specific.h
@@ -17,7 +17,7 @@
 #ifndef __MALI_OSK_SPECIFIC_H__
 #define __MALI_OSK_SPECIFIC_H__
 
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <linux/platform_device.h>
 #include <linux/gfp.h>
 #include <linux/hardirq.h>
diff --git a/src/devicedrv/mali/linux/mali_ukk_core.c b/src/devicedrv/mali/linux/mali_ukk_core.c
index a6c43b7..862beed 100755
--- a/src/devicedrv/mali/linux/mali_ukk_core.c
+++ b/src/devicedrv/mali/linux/mali_ukk_core.c
@@ -9,7 +9,7 @@
  */
 #include <linux/fs.h>       /* file system operations */
 #include <linux/slab.h>     /* memort allocation functions */
-#include <asm/uaccess.h>    /* user space access */
+#include <linux/uaccess.h>    /* user space access */
 
 #include "mali_ukk.h"
 #include "mali_osk.h"
diff --git a/src/devicedrv/mali/linux/mali_ukk_gp.c b/src/devicedrv/mali/linux/mali_ukk_gp.c
index d7c5d2f..bd2bb28 100755
--- a/src/devicedrv/mali/linux/mali_ukk_gp.c
+++ b/src/devicedrv/mali/linux/mali_ukk_gp.c
@@ -8,7 +8,7 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 #include <linux/fs.h>       /* file system operations */
-#include <asm/uaccess.h>    /* user space access */
+#include <linux/uaccess.h>    /* user space access */
 
 #include "mali_ukk.h"
 #include "mali_osk.h"
diff --git a/src/devicedrv/mali/linux/mali_ukk_mem.c b/src/devicedrv/mali/linux/mali_ukk_mem.c
index 3dfe345..a700edf 100755
--- a/src/devicedrv/mali/linux/mali_ukk_mem.c
+++ b/src/devicedrv/mali/linux/mali_ukk_mem.c
@@ -8,7 +8,7 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 #include <linux/fs.h>       /* file system operations */
-#include <asm/uaccess.h>    /* user space access */
+#include <linux/uaccess.h>    /* user space access */
 
 #include "mali_ukk.h"
 #include "mali_osk.h"
diff --git a/src/devicedrv/mali/linux/mali_ukk_pp.c b/src/devicedrv/mali/linux/mali_ukk_pp.c
index 2c59bc7..37ecf40 100755
--- a/src/devicedrv/mali/linux/mali_ukk_pp.c
+++ b/src/devicedrv/mali/linux/mali_ukk_pp.c
@@ -8,7 +8,7 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 #include <linux/fs.h>       /* file system operations */
-#include <asm/uaccess.h>    /* user space access */
+#include <linux/uaccess.h>    /* user space access */
 
 #include "mali_ukk.h"
 #include "mali_osk.h"
diff --git a/src/devicedrv/mali/linux/mali_ukk_profiling.c b/src/devicedrv/mali/linux/mali_ukk_profiling.c
index 2487bf3..2fa3c93 100755
--- a/src/devicedrv/mali/linux/mali_ukk_profiling.c
+++ b/src/devicedrv/mali/linux/mali_ukk_profiling.c
@@ -8,7 +8,7 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 #include <linux/fs.h>       /* file system operations */
-#include <asm/uaccess.h>    /* user space access */
+#include <linux/uaccess.h>    /* user space access */
 #include <linux/slab.h>
 
 #include "mali_ukk.h"
diff --git a/src/devicedrv/mali/linux/mali_ukk_soft_job.c b/src/devicedrv/mali/linux/mali_ukk_soft_job.c
index beeb753..881b751 100755
--- a/src/devicedrv/mali/linux/mali_ukk_soft_job.c
+++ b/src/devicedrv/mali/linux/mali_ukk_soft_job.c
@@ -8,7 +8,7 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 #include <linux/fs.h>       /* file system operations */
-#include <asm/uaccess.h>    /* user space access */
+#include <linux/uaccess.h>    /* user space access */
 
 #include "mali_ukk.h"
 #include "mali_osk.h"
diff --git a/src/devicedrv/mali/linux/mali_ukk_timeline.c b/src/devicedrv/mali/linux/mali_ukk_timeline.c
index f32d8b0..5b0db95 100755
--- a/src/devicedrv/mali/linux/mali_ukk_timeline.c
+++ b/src/devicedrv/mali/linux/mali_ukk_timeline.c
@@ -8,7 +8,7 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 #include <linux/fs.h>       /* file system operations */
-#include <asm/uaccess.h>    /* user space access */
+#include <linux/uaccess.h>    /* user space access */
 
 #include "mali_ukk.h"
 #include "mali_osk.h"
diff --git a/src/devicedrv/mali/linux/mali_ukk_vsync.c b/src/devicedrv/mali/linux/mali_ukk_vsync.c
index 5b54fb9..6cde29c 100755
--- a/src/devicedrv/mali/linux/mali_ukk_vsync.c
+++ b/src/devicedrv/mali/linux/mali_ukk_vsync.c
@@ -8,7 +8,7 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 #include <linux/fs.h>       /* file system operations */
-#include <asm/uaccess.h>    /* user space access */
+#include <linux/uaccess.h>    /* user space access */
 
 #include "mali_ukk.h"
 #include "mali_osk.h"
diff --git a/src/devicedrv/ump/linux/ump_kernel_linux.c b/src/devicedrv/ump/linux/ump_kernel_linux.c
index 32a42ca..023bb11 100755
--- a/src/devicedrv/ump/linux/ump_kernel_linux.c
+++ b/src/devicedrv/ump/linux/ump_kernel_linux.c
@@ -13,7 +13,7 @@
 #include <linux/cdev.h>              /* character device definitions */
 #include <linux/ioport.h>            /* request_mem_region */
 #include <linux/mm.h>                /* memory management functions and types */
-#include <asm/uaccess.h>             /* user space access */
+#include <linux/uaccess.h>           /* user space access */
 #include <asm/atomic.h>
 #include <linux/device.h>
 #include <linux/debugfs.h>
diff --git a/src/devicedrv/ump/linux/ump_osk_low_level_mem.c b/src/devicedrv/ump/linux/ump_osk_low_level_mem.c
index ef35052..d71dd32 100755
--- a/src/devicedrv/ump/linux/ump_osk_low_level_mem.c
+++ b/src/devicedrv/ump/linux/ump_osk_low_level_mem.c
@@ -24,9 +24,9 @@
 #include <linux/kernel.h>
 #include <linux/mm.h>
 #include <linux/slab.h>
+#include <linux/uaccess.h>                        /* to verify pointers from user space */
 
 #include <asm/memory.h>
-#include <asm/uaccess.h>                        /* to verify pointers from user space */
 #include <asm/cacheflush.h>
 #include <linux/dma-mapping.h>
 
diff --git a/src/devicedrv/ump/linux/ump_ukk_ref_wrappers.c b/src/devicedrv/ump/linux/ump_ukk_ref_wrappers.c
index 4a58e93..6981cdb 100755
--- a/src/devicedrv/ump/linux/ump_ukk_ref_wrappers.c
+++ b/src/devicedrv/ump/linux/ump_ukk_ref_wrappers.c
@@ -14,7 +14,7 @@
  */
 
 
-#include <asm/uaccess.h>             /* user space access */
+#include <linux/uaccess.h>             /* user space access */
 
 #include "ump_osk.h"
 #include "ump_uk_types.h"
diff --git a/src/devicedrv/ump/linux/ump_ukk_wrappers.c b/src/devicedrv/ump/linux/ump_ukk_wrappers.c
index 2a88f50..f19e5ec 100755
--- a/src/devicedrv/ump/linux/ump_ukk_wrappers.c
+++ b/src/devicedrv/ump/linux/ump_ukk_wrappers.c
@@ -13,7 +13,7 @@
  * Defines the wrapper functions which turn Linux IOCTL calls into _ukk_ calls
  */
 
-#include <asm/uaccess.h>             /* user space access */
+#include <linux/uaccess.h>             /* user space access */
 
 #include "ump_osk.h"
 #include "ump_uk_types.h"
diff --git a/src/devicedrv/umplock/umplock_driver.c b/src/devicedrv/umplock/umplock_driver.c
index ccd96d3..f20d214 100755
--- a/src/devicedrv/umplock/umplock_driver.c
+++ b/src/devicedrv/umplock/umplock_driver.c
@@ -14,7 +14,7 @@
 #include <linux/slab.h>
 #include <linux/cdev.h>
 #include <linux/device.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include "umplock_ioctl.h"
 #include <linux/sched.h>
 
-- 
2.7.4

